sd-core/src/language/llvm_ir.rs

```diff
@@ -0,0 +1,670 @@
+                Instruction::Or(or) => vec![or.operand0.into(), or.operand1.into()],
+                Instruction::Xor(xor) => vec![xor.operand0.into(), xor.operand1.into()],
+                Instruction::Shl(shl) => vec![shl.operand0.into(), shl.operand1.into()],
+                Instruction::LShr(lshr) => vec![lshr.operand0.into(), lshr.operand1.into()],
+                Instruction::AShr(ashr) => vec![ashr.operand0.into(), ashr.operand1.into()],
+                Instruction::FAdd(fadd) => vec![fadd.operand0.into(), fadd.operand1.into()],
+                Instruction::FSub(fsub) => vec![fsub.operand0.into(), fsub.operand1.into()],
+                Instruction::FMul(fmul) => vec![fmul.operand0.into(), fmul.operand1.into()],
+                Instruction::FDiv(fdiv) => vec![fdiv.operand0.into(), fdiv.operand1.into()],
+                Instruction::FRem(frem) => vec![frem.operand0.into(), frem.operand1.into()],
+                Instruction::FNeg(fneg) => vec![fneg.operand.into()],
+                Instruction::ExtractElement(extract_element) => {
+                    vec![extract_element.vector.into(), extract_element.index.into()]
+                }
+                Instruction::InsertElement(insert_element) => vec![
+                    insert_element.vector.into(),
+                    insert_element.element.into(),
+                    insert_element.index.into(),
+                ],
+                Instruction::ShuffleVector(shuffle_vector) => vec![
+                    shuffle_vector.operand0.into(),
+                    shuffle_vector.operand1.into(),
+                ],
+                Instruction::ExtractValue(extract_value) => [
+                    vec![extract_value.aggregate.into()],
+                    extract_value
+                        .indices
+                        .into_iter()
+                        .map(|idx| {
+                            Operand::ConstantOperand(ConstantRef::new(Constant::Int {
+                                bits: 32,
+                                value: idx.into(),
+                            }))
+                            .into()
+                        })
+                        .collect(),
+                ]
+                .concat(),
+                Instruction::InsertValue(insert_value) => [
+                    vec![insert_value.element.into()],
+                    insert_value
+                        .indices
+                        .into_iter()
+                        .map(|idx| {
+                            Operand::ConstantOperand(ConstantRef::new(Constant::Int {
+                                bits: 32,
+                                value: idx.into(),
+                            }))
+                            .into()
+                        })
+                        .collect(),
+                ]
+                .concat(),
+                Instruction::Alloca(alloca) => vec![
+                    /* TODO: handle type */
```

Comment:
Could you comment on the TODOs? what's missing, and how important is it?

Reply:
It's extra information exposed by `llvm-ir`, like calling conventions, attributes, etc.
Basically, metadata (including type information of operands) which we have no way of rendering now (if ever).